compute/standard-deviation/docs/modules/ROOT/pages/compute-standard-deviation.adoc

```diff
@@ -0,0 +1,332 @@
+----
+
+This code is easy to understand, which is good for maintenance. However, it has three flaws:
+
+- This code requires every customer record to be moved from where the data is stored (in this case the Hazelcast cluster) to where the calculation is run. Even if a projection were added, this is very heavy on the network.
+
+- The `keySet()` operation produces a collection that contains all the keys to iterate across. If this collection is large, it could result in memory overflow.
+
+- If the map is empty, the code results in a division by zero. A good unit test would find this.
+
+The sum of the square of the differences is calculated on the client-side like this:
+
+[source,java,indent=0]
+----
+include::standard-deviation:example$client/src/main/java/your/company/name/JuniorDeveloper.java[tag=stdev]
+----
+
+link:{github-directory}/client/src/main/java/your/company/name/JuniorDeveloper.java[View source]
+====
+
+.Client `SeniorDeveloper.java`
+[%collapsible]
+====
+In this class, the average is calculated in one line:
+
+[source,java,indent=0]
+----
+include::standard-deviation:example$client/src/main/java/your/company/name/SeniorDeveloper.java[tag=average]
+----
+
+Hazelcast provides built-in functions for this type of calculation.
+
+The sum of the square of the differences is calculated on the cluster-side like this:
+
+[source,java,indent=0]
+----
+include::standard-deviation:example$client/src/main/java/your/company/name/SeniorDeveloper.java[tag=stdev]
+----
+
+The client uses the ExecutorService API to run a `Callable` task on each member in the cluster. The `Callable` class is named `TotalDifferenceSquaredCallable.java` and it runs the calculation on the Hazelcast member where it is invoked.
+
+Each member calculates a sub-total only for the data that member owns. If you have two members, the calculation takes half the time. If you have ten members, the calculation takes a tenth of the time. Each member runs it's calculation independently, the runtime is dependent on how much data each member hosts rather than how much data as a whole exists.
+
+To do this, the task uses the `localKeySet()` method instead of the `keySet()` method. This `localKeySet()` method returns the keys that are held by the current process, and since the task is run on all processes, all the keys are included.
+
+Each member in the cluster only needs to work with the keys, and therefore entries, that it owns. As a result, members don't have to do any network calls to retrieve the keys from elsewhere.
+
+Finally, the task is submitted across each member in the cluster, which results in a collection of `Future` objects. One `Future` is returned for each member that runs the task, and the task needs to wait for them all to finish. This is not exactly difficult, and since all members will have roughly the same amount of data, you can assume execution time is equal and iterate across this collection running `Future.get()` to get the sub-total from each member.
```

Comment:
```suggestion
Finally, the task is submitted across each member in the cluster, which results in a collection of `Future` objects. One `Future` is returned for each member that runs the task, and the task needs to wait for them all to finish. This is not exactly difficult, since all members will have roughly the same amount of data. You can assume that the execution time is the same for all members and iterate across this collection running `Future.get()` to get the subtotal from each member.
```